### DISCUSSION TITLE
Toggle Container Runtime - Descriptions and Wording

### DISCUSSION THREAD

**Discussion by rak-phillip** [rancher-sandbox/rancher-desktop#915] [Q&A]
#900 will introduce the ability to select a container runtime on first launch of Rancher Desktop. This decision can be perceived as a big one, especially to new users who might be trying out Rancher Desktop for the first time; other's might not might not be aware of the difference between  `dockerd` and `containerd` and wont necessarily know the benefits of choosing one over the other. 

Because this has potential to be such a confusing option on first launch, I want to make sure that we have effective messaging that will help people understand the decision they are making. There are at least two problems that we need to address:

1. Make sure that labels for our options are accurate and clear (Currently `Container Runtime`, with two options, `dockerd (moby)` & `containerd`
2. Provide content for a tooltip that will explain the difference between the two and why one would want to pick one option over the other. (I'm also considering that we might want to emphasize that this can be changed later)

### Making a selection
![11-Welcome](https://user-images.githubusercontent.com/835961/140581452-2ad4235d-39eb-4a18-8952-e3d4b6126a80.png)

### Tooltip active
![11-Welcome – 1](https://user-images.githubusercontent.com/835961/140581448-b0eb79cb-7378-4774-b6e6-7fda5dfb3247.png)

**Comment by mook-as** [comment]
It may be better to provide option-specific help rather than help for the whole section:

- dockerd (moby)
  - Provides better compatibility with Docker tooling.
- containerd
  - Provides better compatibility with Kubernetes-native applications.

**Reply by jandubois** [reply]
The description should tell the user the actual benefit they get, and not just describe the feature. It has to answer the questions: "Why do I want it? What can I do with it?"

> * dockerd (moby)
>   
>   * Provides better compatibility with Docker tooling.

* Provide native docker API; use with docker cli and k3d.

> * containerd
>   
>   * Provides better compatibility with Kubernetes-native applications.

What actually is the "better compatibility"? We could say "less resources", but we need to measure first to know if this is noticeable.

* Separate namespaces for regular and kubernetes container images; use with nerdctl.

Not really happy with these either, but they show the direction I want to change to.

**Reply by rak-phillip** [reply]
Updated the mockups to reflect the feedback. Reduced the descriptions so that they easily fit on a single line. How do you feel about this change compared to what we originally had above?

### Making a selection
![11-Welcome](https://user-images.githubusercontent.com/835961/141013045-f53f74ce-a7c7-4d1b-bd5d-79912ab2ef53.png)

@mook-as @jandubois

**Reply by mook-as** [reply]
CLI is an initialism ("command line interface") and should be all upper case.  The tools themselves are typically lower case (and so they're fine as-is).

For containerd, I was initially thinking that other Kubernetes-focused tooling are more likely to deal with containerd (e.g. kim); however, maybe that's a special case and people normally don't use things that go that low-level…

**Reply by jandubois** [reply]
I don't know what "native" means here; I think it is redundant. Either you support the docker API, or you don't.

**Reply by rak-phillip** [reply]
Thanks! Updated to reflect the feedback

![11-Welcome](https://user-images.githubusercontent.com/835961/141032243-eda3ed34-96ec-422a-bda4-5c10a5d5bf0a.png)
